views/005_component_views.sql

```diff
@@ -147,23 +147,18 @@ CREATE OR REPLACE VIEW component_types AS
   FROM components ORDER BY type asc;
 
 CREATE OR REPLACE FUNCTION lookup_component_config_id_related_components (
-  component_id TEXT,
-  type_filter TEXT DEFAULT 'outgoing'
+  component_id TEXT
 )
 RETURNS TABLE (id UUID) AS $$
 BEGIN
     RETURN QUERY
-        WITH related_config_ids AS (
-            SELECT * 
-            FROM related_configs_recursive(
-                (SELECT config_id FROM components WHERE components.id = $1::UUID), 
-                $2, 
-                FALSE, 
-                5
-            )
+        WITH config_id_paths AS (
+            SELECT config_items.id
+            FROM config_items
+            WHERE path LIKE '%' || (SELECT config_id FROM components WHERE components.id = $1::UUID) || '%'
         )
         SELECT components.id 
         FROM components 
-        WHERE config_id IN (SELECT related_config_ids.id FROM related_config_ids);
```

Comment:
I think a join will be faster: https://dba.stackexchange.com/questions/91247/optimizing-a-postgres-query-with-a-large-in

Reply:
Done 👍🏽